### DISCUSSION TITLE
BPP implementation: how to update ride status to the invoking BAP

### DISCUSSION THREAD

**Discussion by sudhirrg** [beckn/mobility#41] [Q&A]
In the ride hailing application,once the driver completes the ride and receives the payment, the ride gets ended, the beckn transaction also ends. (a) How do we update the status (e.g. 'Ride ended') to bap? 
   (b) How do we update the payment status in the Payment object?

**Comment by sjthnrk** [comment]
I’m not sure if the concept of beckn transaction ending with ride is a way to look at it.  A ride transaction covering pre booking, on ride and post ride can be broken to numerous beckn api calls and it’s upto the implementir on how the ride system of the implementor applies the beckn calls. For example, there could be several on-update or on_statuscalls that can be made even after ride ends (to update final price of ride, or other ride specific data) after ride ends. 

responses to your Qs with this background


A) an independent on_status call can be made by BPP for the same `transaction_id`.  BAP could give an ack if it’s the first time and choose to ignore all follow on_status calls carry same status update

B) using On_status call again

**Comment by ravi-prakash-v** [comment]
Whenever the BPP needs to send a status update to the BAP, it must call the `on_status` endpoint of the BAP. For a ride hailing BPP, typical status updates that can be sent are
- Searching for Driver
- Driver Assigned
- Driver on the way to pickup
- Driver arrived at pickup
- Ride Started
- Ride ended

There could be more. 

Each of these status updates can be sent via an unsolicited `on_status` call from the BPP to the BAP
The field that will be updated in the on_status call is the `message.order.fulfillment.status.descriptor.name` and `message.order.fulfillment.status.descriptor.code`

Hope that helps.
